### ISSUE TITLE
Action status is "succeeded" even though it failed

### ISSUE THREAD

**Issue by MaxDesiatov** [NSHipster/update-homebrew-formula-action#3]
See this run https://github.com/swiftwasm/carton/runs/1368971850?check_suite_focus=true

It failed with

```
F, [2020-11-08T00:33:20.150543 #1] FATAL -- : PUT https://api.github.com/repos/swiftwasm/homebrew-tap/contents/carton.rb: 403 - Resource not accessible by integration // See: https://docs.github.com/rest/reference/repos#create-or-update-file-contents (Octokit::Forbidden)
/usr/local/bundle/gems/octokit-4.18.0/lib/octokit/response/raise_error.rb:16:in `on_complete'
/usr/local/bundle/gems/faraday-1.1.0/lib/faraday/response.rb:12:in `block in call'
/usr/local/bundle/gems/faraday-1.1.0/lib/faraday/response.rb:65:in `on_complete'
/usr/local/bundle/gems/faraday-1.1.0/lib/faraday/response.rb:11:in `call'
/usr/local/bundle/gems/octokit-4.18.0/lib/octokit/middleware/follow_redirects.rb:73:in `perform_with_redirection'
/usr/local/bundle/gems/octokit-4.18.0/lib/octokit/middleware/follow_redirects.rb:61:in `call'
/usr/local/bundle/gems/faraday-1.1.0/lib/faraday/request/retry.rb:148:in `call'
/usr/local/bundle/gems/faraday-1.1.0/lib/faraday/rack_builder.rb:154:in `build_response'
/usr/local/bundle/gems/faraday-1.1.0/lib/faraday/connection.rb:492:in `run_request'
/usr/local/bundle/gems/faraday-1.1.0/lib/faraday/connection.rb:279:in `put'
/usr/local/bundle/gems/sawyer-0.8.2/lib/sawyer/agent.rb:94:in `call'
/usr/local/bundle/gems/octokit-4.18.0/lib/octokit/connection.rb:156:in `request'
/usr/local/bundle/gems/octokit-4.18.0/lib/octokit/connection.rb:37:in `put'
/usr/local/bundle/gems/octokit-4.18.0/lib/octokit/client/contents.rb:83:in `create_contents'
/usr/local/bundle/gems/octokit-4.18.0/lib/octokit/client/contents.rb:116:in `update_contents'
/entrypoint.rb:138:in `<main>'
```

but the action run itself is marked as "succeeded".

**Comment by MaxDesiatov** [comment]
Speaking of the failure itself, I guess this action doesn't work without a PAT after all?

**Comment by MaxDesiatov** [comment]
It doesn't work in this repo with a PAT either https://github.com/swiftwasm/carton/runs/1373800421?check_suite_focus=true

**Comment by mattt** [comment]
Sorry this action keeps spoiling your releases. I really wish there was a reasonable way to dry run and end-to-end test these integrations.

I vaguely recall getting this error myself over the course of development, but I don't remember the exact cause or how I resolved it. Regardless, the fact that this is succeeding despite the errors should definitely be fixed. Looking into that now.

**Comment by MaxDesiatov** [comment]
No worries, thanks for your help! I know this is hard to test 🙂

**Comment by mattt** [comment]
@MaxDesiatov This should be fixed by #4.

As for the underlying issue, I think `GITHUB_TOKEN` may indeed have insufficient privileges, so you'll need to pass a generated PAT (similar to what's necessary for https://github.com/SwiftDocOrg/github-wiki-publish-action). I'll test this out and open a PR with the appropriate documentation changes.

**Comment by mattt** [comment]
@MaxDesiatov See #6.
